django/db/models/fields/related.py

```diff
@@ -464,22 +464,33 @@ def _check_unique_target(self):
         if not self.foreign_related_fields:
             return []
 
-        has_unique_field = any(rel_field.unique
-            for rel_field in self.foreign_related_fields)
-        if not has_unique_field and len(self.foreign_related_fields) > 1:
+        unique_foreign_fields = {
+            frozenset([f.name])
+            for f in self.remote_field.model._meta.get_fields()
+            if getattr(f, 'unique', False)
+        }
+        unique_foreign_fields.update({
+            frozenset(ut)
+            for ut in self.remote_field.model._meta.unique_together
+        })
+        foreign_fields = {f.name for f in self.foreign_related_fields}
+
+        has_unique_constraint = any(u <= foreign_fields for u in unique_foreign_fields)
+
+        if not has_unique_constraint and len(self.foreign_related_fields) > 1:
             field_combination = ', '.join("'%s'" % rel_field.name
                 for rel_field in self.foreign_related_fields)
             model_name = self.remote_field.model.__name__
             return [
                 checks.Error(
-                    "None of the fields %s on model '%s' have a unique=True constraint."
-                    % (field_combination, model_name),
+                    "None of the fields %s on model '%s' have a unique=True or "
+                    "unique_together constraint." % (field_combination, model_name),
```

Comment:
They don't need to be *all* included in a unique constraint, a subset of them can be in a `unique_together` constraint. For example, it can point to `(a, b, c)` and only `(a, b)` can be `unique_together`.

But I understand your point though :) . I'll rephrase this, thanks.